server/app/views/components/PageNotProductionBanner.java

```diff
@@ -0,0 +1,89 @@
+package views.components;
+
+import static com.google.gdata.util.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.h4;
+import static j2html.TagCreator.join;
+import static j2html.TagCreator.p;
+
+import com.google.inject.Inject;
+import j2html.tags.specialized.ATag;
+import j2html.tags.specialized.DivTag;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.mvc.Http;
+import services.DeploymentType;
+import services.settings.SettingsManifest;
+
+/**
+ * This will build a sticky banner that follow the user as they scroll. It will show only on a
+ * staging site that has the CIVIC_ENTITY_PRODUCTION_URL setting configured.
+ *
+ * <p>The purpose is to make it extremely clear to people loading a page if the site is Production
+ * or not.
+ */
+public final class PageNotProductionBanner {
+
+  private static final Logger logger = LoggerFactory.getLogger(PageNotProductionBanner.class);
+  private final SettingsManifest settingsManifest;
+  private final DeploymentType deploymentType;
+
+  @Inject
+  public PageNotProductionBanner(SettingsManifest settingsManifest, DeploymentType deploymentType) {
+    this.settingsManifest = checkNotNull(settingsManifest);
+    this.deploymentType = checkNotNull(deploymentType);
+  }
+
+  public Optional<DivTag> render(Http.Request request) {
+    if (deploymentType.isDev()) {
```

Comment:
**suggestion:** It might be nice to allow the banner to be shown on dev as well. If the reason we don't want to do this is because of the browser tests, could we just not provide the URL there and it wouldn't show?

Reply:
Good point